packages/Telephony/Actions/SerialNumberInput.cs

```diff
@@ -142,56 +164,80 @@ public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext d
                         return await PromptUserAsync(dc, cancellationToken).ConfigureAwait(false);
                 }
 
-                if (snp.PatternLength == result.Length)
+                if (batchLength == result.Length)
                 {
                     //If so, save it to the output property and end the dialog
                     //Get property from expression
-                    string property = this.Property?.GetValue(dc.State);
+                    var property = Property?.GetValue(dc.State);
                     dc.State.SetValue(property, result);
                     return await dc.EndDialogAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
                 }
                 else
                 {
                     dc.State.SetValue(AggregationDialogMemory, result);
-                    await this.SendActivityMessageAsync(this.ContinuePrompt, dc, cancellationToken).ConfigureAwait(false);
+                    await SendActivityMessageAsync(ContinuePrompt, dc, cancellationToken).ConfigureAwait(false);
                     return new DialogTurnResult(DialogTurnStatus.Waiting);
                 }
             }
-            
-            //append the message to the aggregation memory state
+
+            //If input is longer than pattern length and wasn't handled as an interruption, silently disregard input.
+            var userInput = dc.Context.Activity.Text;
+
+            if (userInput.Length > batchLength)
+            {
+                return new DialogTurnResult(DialogTurnStatus.Waiting);
+            }
+
+            //Append the message to the aggregation memory state and check if its length is longer than batch length.
             var existingAggregation = dc.State.GetValue(AggregationDialogMemory, () => string.Empty);
-            existingAggregation += dc.Context.Activity.Text;
 
-            string[] results = snp.Inference(existingAggregation);
+            //Example happy path: PatternLength is 5, AcceptsAlphabet and AcceptsDigits
+            //If user says "A", "B", "C", "1", "22", do not update state as "ABC122" is invalid. Wait for them to provide valid input.
+            if ((existingAggregation + userInput).Length > batchLength)
+            {
+                return new DialogTurnResult(DialogTurnStatus.Waiting);
+            }
+
+            //TODO: Turn counter? How long is someone in this node?
```

Comment:
why not add the logic now?
you could always compare with a reasonably high constant?

Reply:
Some kind of circuit-breaker is needed here. For fixed length input you can count the size of input entered so far and get out if that exceeds the expected length by some number. For pound-terminated sequences you can rely on the user pressing #.

Reply:
After thinking it through, I'm putting together a circuit breaker that is configurable via the `"unexpectedInputsUntilInterruptionsAreAllowed"` property.

It defaults to a value of **2**, and is only respected when `"allowInterruptions"` is **false**.

Here's an example from the docstring for the new property:

    /// E.g. BatchLength is 5, AllowInterruptions is false and UnexpectedInputsUntilInterruptionsAreAllowed is 2.
    /// When a user says "cancel" three times, the third "cancel" will be handled as an interruption.

C2 Users do not need to say the same utterance until it is handled as an interruption.

Using the same configuration above, the user could sequentially say "cancel", "quit", and "exit", and after "quit", "exit" would be handled as an interruption.

Any zero or negative value for `"unexpectedInputsUntilInterruptionsAreAllowed"` will be ignored and the default value of **2** will be used, and _ideally_ a LogWarning would occur. But logging doesn't seem to be a pattern inside of `Microsoft.Bot.Dialogs.Adaptive` or in `Microsoft.Bot.Components.Telephony`, so maybe an error is thrown at runtime 😢 🤷🏼